ibis/backends/snowflake/__init__.py

```diff
@@ -221,7 +234,65 @@ def normalize_name(name):
                 return name
 
         self.con.dialect.normalize_name = normalize_name
-        return res
+        self.database_name = self._default_database = dbparams["database"]
+        self._default_schema = dbparams["schema"]
+        return eng
+
+    @staticmethod
+    def from_snowpark(session) -> Backend:
+        """Create an Ibis Snowflake backend from a Snowpark session.
+
+        Parameters
+        ----------
+        session
+            A snowpark session instance.
+
+        Returns
+        -------
+        Backend
+            An Ibis Snowflake backend instance.
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> import snowflake.snowpark as sp  # doctest: +SKIP
+        >>> session = sp.Session.builder.configs(...).create()  # doctest: +SKIP
+        >>> con = ibis.snowflake.from_snowpark(session)  # doctest: +SKIP
+        >>> batting = con.tables.BATTING  # doctest: +SKIP
+        >>> batting[["playerID", "RBI"]].head()  # doctest: +SKIP
+        ┏━━━━━━━━━━━┳━━━━━━━┓
+        ┃ playerID  ┃ RBI   ┃
+        ┡━━━━━━━━━━━╇━━━━━━━┩
+        │ string    │ int64 │
+        ├───────────┼───────┤
+        │ abercda01 │     0 │
+        │ addybo01  │    13 │
+        │ allisar01 │    19 │
+        │ allisdo01 │    27 │
+        │ ansonca01 │    16 │
+        └───────────┴───────┘
+        """
+        import ibis
+
+        con = session._conn._conn
+
+        def creator():
+            # workaround a broken implementation of the DBAPI cursor
+            # when con._paramstyle is **not** set to pyformat
+            #
+            # when using snowpark, con._paramstyle is set to qmark
+            con._paramstyle = "pyformat"
+            return con
+
+        return ibis.connect(
+            "snowflake://",
+            creator=creator,
+            user=con.user,
+            password=None,
+            database=f"{con.database}/{con.schema}",
+            account=con.host.replace(".snowflakecomputing.com", ""),
```

Comment:
Rather than do this, I think you can call [get_current_account()](https://github.com/snowflakedb/snowpark-python/blob/b50e4fcf4e2875eb1d392253fdaf0b8c117ee215/src/snowflake/snowpark/session.py#L1705).

Reply:
Unfortunately that doesn't include the cloud provider, just the `xx12345` part.

Reply:
Should it include the extra stuff in this case? I suppose it depends on what ibis is doing with it. The other part is part of the url but I don't think of it as an account property. 

¯\\\_(ツ)_/¯

Reply:
The only thing we definitely need is the `database=` argument, so maybe I'll just drop the rest.

Reply:
Fixed, removed the unnecessary stuff.